python/cudf/cudf/core/dataframe.py

```diff
@@ -783,11 +783,23 @@ def fallback(col, fn):
                 else:
                     result[col] = fallback(rhs._cols[col], _reverse_op(fn))
         elif isinstance(other, Series):
-            raise NotImplementedError(
-                "Series to DataFrame arithmetic not supported "
-                "until strings can be used as indices. Try converting your"
-                " Series into a DataFrame first."
-            )
+            result_cols = list(self.columns)
+            for new_col in list(other.index):
+                if new_col not in result_cols:
+                    result_cols.append(new_col)
+            for col in result_cols:
+                if col in self.columns and col in other.index:
+                    l_opr = self._cols[col]
+                    r_opr = other[other.index == col][0]
+                else:
+                    if col not in self.columns:
+                        l_opr = Series([np.nan] * len(self))
```

Comment:
Don't create the empty column this way, this will create a big Python list, then convert it into something like a numpy array / Pandas Series, and then copy to the GPU, then run a kernel to convert `NaN` to nulls. We should be able to directly construct empty columns of a given size using some of our column functions, while also maintaining the dtype of `other`.